src/card/storage_adapters/git_storage_adapter.rb

```diff
@@ -1,22 +1,46 @@
+require_relative "../models.rb"
+require "open-uri"
+
 module Card
   module StorageAdapters
     # Saves the images in the local git repository
     class GitStorageAdapter < Abstract
       # todo move to config
       ASSET_PATH = "assets/core/card/printed"
+      PRINT_MANIFEST_PATH = "assets/core/card/printed/print_manifest.json"
+      GH_URL = "https://raw.githubusercontent.com/elliottomlinson/rpcg"
+
       def save(cards)
         cards.each do |card|
           FileUtils.cp(
             card.image_path,
-            card_file_name(card.spec)
+            card_file_name(card)
           )
         end
+
+        update_manifest(cards)
+      end
+
+      def printed_cards
+        raise "Git Print Manifest unavailable, try regenerating assets" unless File.exists?(PRINT_MANIFEST_PATH)
+
+        Marshal.load(File.read(PRINT_MANIFEST_PATH)).each do |printed_card|
+          printed_card.image_url = github_repo_url(printed_card)
+        end
       end
 
       private
 
-      def card_file_name(card_spec)
-        File.join(ASSET_PATH, "#{card_spec.title}.png")
+      def update_manifest(cards)
+        File.write(PRINT_MANIFEST_PATH, Marshal.dump(cards))
+      end
```

Comment:
`Marshal` lets us serialize/deserialize objects to and from files. This is useful in our two step print/import process that requires us to get our cards onto the master branch before we can import.

Reply:
This way you generate the cards and have their metadata added to the manifest, push all the information into GH, then anybody who wants to import the cards into Tabletop can just run the import script.